src/libraries/System.Security.Cryptography.Algorithms/tests/ChaCha20Poly1305Tests.cs

```diff
@@ -447,6 +447,10 @@ public static void CheckIsSupported()
                 // The test queries the OS directly to ensure our version check is correct.
                 expectedIsSupported = CngUtility.IsAlgorithmSupported("CHACHA20_POLY1305");
             }
+            else if (PlatformDetection.IsAndroid)
+            {
+                expectedIsSupported = true;
```

Comment:
We should have this check if it's supported at runtime so the unit tests will still pass when run on older supported simulator versions.

Reply:
I didn't actually see a good way to check that `IsSupported` is implemented sanely on Android. I don't see an API anywhere that tells you the API Level (sdk version) of the current version of Android.

It doesn't make sense to check this the same way that `IsSupported` is implemented (seeing if we can instantiate a ChaCha20/Poly1305/NoPadding cipher), since we'd essentially be doing `Assert.Equal(ChaChaPoly1305.IsSupported, ChaChaPoly1305.IsSupported)`.

`OperatingSystem.IsAndroidVersionAtLeast` doesn't help us here, either. It depends on the Linux kernel version, not the OS version (and sometimes the API Level bumps for the same OS via OTA updates). 

I didn't see `android.os.Build.VERSION.SDK_INT` exposed in a manged API, either. So: I ended up shelling out to `getprop`, for now.

Again, definitely open to suggestions for improvements or ideas.